library/extractor/src/main/java/com/google/android/exoplayer2/text/dvb/DvbTeletextParser.java

```diff
@@ -0,0 +1,513 @@
+package com.google.android.exoplayer2.text.dvb;
+
+import android.os.Build;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+import android.text.style.BackgroundColorSpan;
+import android.text.style.ForegroundColorSpan;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.android.exoplayer2.text.Cue;
+import com.google.android.exoplayer2.util.Log;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** Parses {@link Cue}s from a DVB Teletext subtitle bitstream. */
+/* package */ final class DvbTeletextParser {
+
+  private static final String TAG = "DvbTeletextParser";
+
+  private static final int TTX_HEIGHT = 576;
+  private static final int TTX_COLS = 40;
+  private static final int TTX_ROWS = 24;
+  private static final int CHAR_HEIGHT = 21;
+
+  private static final int[] TXT_COLORS = {
```

Comment:
Don't use abbreviations like TXT unless it's a literal reference to something in the spec

Reply:
Done in 69f49d5388fbdc6972d7ca7517171fad3617d4de